traffic_portal/app/src/common/modules/dialog/select/lock/dialog.select.lock.tpl.html

```diff
@@ -0,0 +1,69 @@
+</div>
+<div class="modal-body">
+    <form name="lockForm" novalidate>
+        <div class="row">
+            <div class="col-sm-12 col-md-12">
+                <div class="form-group" ng-class="{'has-error': hasError(lockForm.cdn), 'has-feedback': hasError(lockForm.cdn)}">
+                    <label class="control-label" for="cdn">CDN *
+                        <small class="input-error" ng-show="hasPropertyError(lockForm.cdn, 'required')">Required</small>
+                    </label>
+                    <select id="cdn" name="cdn" class="form-control" ng-model="lock.cdn" ng-options="cdn.name as cdn.name for cdn in cdns" required>
+                        <option value="">Select...</option>
+                    </select>
+                </div>
+            </div>
+        </div>
+        <div class="row">
+            <div class="col-sm-12 col-md-12">
+                <div class="form-group" ng-class="{'has-error': hasError(lockForm.message), 'has-feedback': hasError(lockForm.message)}">
+                    <label class="control-label" for="message">Message *
+                        <small class="input-error" ng-show="hasPropertyError(lockForm.message, 'maxlength')">Too Long</small>
+                    </label>
+                    <input id="message" name="message" type="text" class="form-control" ng-model="lock.message" ng-maxlength="256" required autofocus>
+                    <span ng-show="hasError(lockForm.message)" class="form-control-feedback"><i class="fa fa-times"></i></span>
+                </div>
+            </div>
+        </div>
+        <div class="row">
+                <div class="col-sm-12 col-md-12">
+                    <div class="form-group">
+                        <p>Type *</p>
+                        <input id="soft" type="radio" data-ng-model="lock.soft" data-ng-value="true">
+                        <label class="has-tooltip" for="soft">&nbsp;Soft<div class="helptooltip">
+                            <div class="helptext">A soft lock will prevent others from queueing server updates or snapshotting the locked CDN.</div>
+                        </div>
+                        </label><br>
+                        <input id="hard" type="radio" data-ng-model="lock.soft" data-ng-value="false">
+                        <label class="has-tooltip" for="soft">&nbsp;Hard<div class="helptooltip">
+                            <div class="helptext">A hard lock will prevent others from queueing server updates or snapshotting the locked CDN. It will also block others from making changes that would change the state of the locked CDN.</div>
+                        </div>
+                        </label>
+                    </div>
+            </div>
+        </div>
+    </form>
+</div>
+<div class="modal-footer">
+    <button class="btn action-btn" ng-click="cancel()">Cancel</button>
+    <button class="btn btn-link" ng-disabled="lockForm.$invalid" ng-click="select()">Submit</button>
```

Comment:
`button`s that don't submit containing `form`s should be `button[type=button]`s, as the implicit default is `button[type=submit]`.